### PR TITLE
Add step annotate and improve AADHandleConditionalAccessFilter for Conditional Access.

### PR DESCRIPTION
**Pull request by han-gao**
1. Add conditional access annotate steps.
2. Change the conditional access code and use OAuth2AuthorizationRequestRedirectFilter to handle exceptions.
3. Add handling of unauthorized access.
4. Remove useless imports.

### CODE REVIEW

**Comment by chenrujun** [comment]
Hi, @han-gao .
It's better to add description when you create a PR.

File: sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/aad/webapp/AuthorizationClientProperties.java [hunk 1]
```diff
@@ -10,6 +10,8 @@
  */
 public class AuthorizationClientProperties {
 
+    public static final String AZURE_CLIENT_REGISTRATION_ID = "azure";
```
Comment (chenrujun) [inline sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/aad/webapp/AuthorizationClientProperties.java:13]: Seems we already have `AZURE_CLIENT_REGISTRATION_ID` in `AADClientRegistrationRepository`.
Please use that instead.

**Review by chenrujun — CHANGES_REQUESTED** [review]

File: sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/aad/webapp/AuthorizationClientProperties.java (hunk 1 above)
Comment (han-gao) [reply sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/aad/webapp/AuthorizationClientProperties.java:13]: Sorry, I didn't notice that.

File: sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/aad/webapp/AuthorizationClientProperties.java [hunk 2]
```diff
@@ -10,6 +10,8 @@
  */
 public class AuthorizationClientProperties {
 
+    public static final String AZURE = "azure";
```
Comment (chenrujun) [inline sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/aad/webapp/AuthorizationClientProperties.java:13]: I mean delete this one, and use `AADClientRegistrationRepository.AZURE` instead.

**Review by chenrujun — CHANGES_REQUESTED** [review]

File: sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/aad/webapp/AuthorizationClientProperties.java (hunk 2 above)
Comment (han-gao) [reply sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/aad/webapp/AuthorizationClientProperties.java:13]: Got it!

**Review by chenrujun — APPROVED** [review]
LGTM.

**Pull request merged**